File: javascript/leetcode/remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    rootNode = new ListNode(0)
    rootNode.next = head
    prevHead = rootNode
    var lastVal
    while (prevHead.next !== null) {
        if (prevHead.next.val == lastVal) {
            lastVal = prevHead.next.val
            prevHead.next = prevHead.next.next
        } else {
            lastVal = prevHead.next.val
            prevHead = prevHead.next
        }
    }
    return rootNode.next
};

File: python/leetcode/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
	def deleteDuplicates(self, head):
		root_node = ListNode(0)
		root_node.next = head
		prev_head = root_node
		last_val = None
		while prev_head.next is not None:
			if prev_head.next.val == last_val:
				last_val = prev_head.next.val
				prev_head.next = prev_head.next.next
			else:
				last_val = prev_head.next.val
				prev_head = prev_head.next
		return root_node.next